src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStaticTests.cs

```diff
@@ -1178,6 +1178,95 @@ End Function
             }.RunAsync();
         }
 
+        [Fact]
+        public async Task FullProperties_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class C
+{
+    private int field;
+
+    public int P1
+    {
+        get { return field; }
+        set { field = value; }
+    }
+
+    public int P2
+    {
+        get { return field; }
+    }
+
+    public int P3 => field;
+
+    public int P4
+    {
+        get => field;
+        set => field = value;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task AutoProperties_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+public class C1
+{
+    public string P1 { get; set; }
+
+    public string P2 { get; }
+
+    public string P3
+    {
+        [DebuggerStepThrough]
+        get;
+    }
+
+    public string P4
```

Comment:
With the new implementation we will start reporting for `P4` as there is no implicitly declared field for this property (most likely because of the error). I am fine with either keeping the error or working around it by also allowing not to find an associated field for the property. Let me know your preference and I will update the code.

Reply:
Yeah, we don't care if there is a compiler error here, its fine to adjust the test to expect a diagnostic here.